### PR TITLE
node-matter code move

### PR DESCRIPTION
**Pull request by Apollon77**
Here wo go. Let's give a warm welcome to the node-matter codebase to join the matter.js project now as matter-node.js package.

This PR copntains multiple commits:
* Moving Main branch content "as is" from node-matter as of now
* Adjustments needed to make it build and all work
* Another Linter error run (no idea why these were not reported while executing the linter in the old repo)
* Copyright adjustments for all files of node-matter and also year update for the matter.js ones
* Formatting adjustments by running the formatter
* And finally changes because I understood how *.buildinfo works in TS and that we want to have that in our monorepo project, so adding a build-clean to separate from build which should still be the default

After this PR has been merged I will bring in all open PRs from node-matter and more

### CODE REVIEW

File: packages/matter-node.js/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Contributing
```
Comment (mfucci) [inline packages/matter-node.js/CONTRIBUTING.md:1]: Probably this should be moved to the root now.

File: packages/matter-node.js/src/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1 @@
+* @mfucci
```
Comment (mfucci) [inline packages/matter-node.js/src/CODEOWNERS:1]: There can be only one CODEOWNERS file per repository.
Delete this one...

File: packages/matter-node.js/src/codec/BitmapSchema.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
+/**
```
Comment (mfucci) [inline packages/matter-node.js/src/codec/BitmapSchema.ts:1]: This is a replacement for /packages/matter.js/src/schema/BitmapSchema.ts
I forked it to add more features but never merged it back.
Either:
1 _ do the overwrite in the MR
2 _ create another MR to do the overwrite now and I will approve both

File: tsconfig.json [hunk 4]
```diff
@@ -16,5 +16,5 @@
         "noUnusedLocals": true,
         "strictNullChecks": true
     },
-    "exclude": ["**/*.d.ts"]
+    "exclude": ["**/dist/**/*", "node-modules/**/*"]
```
Comment (mfucci) [inline tsconfig.json:19]: "dist/**/*" ?
I think "**/dist/**/*" and "dist/**/*"  are equivalent but this is not consistent with "node-modules/**/*"

**Review by mfucci — CHANGES_REQUESTED** [review]
Can you add in this MR node-matter tests to the CI pipeline?

File: packages/matter-node.js/src/CODEOWNERS (hunk 2 above)
Comment (Apollon77) [reply packages/matter-node.js/src/CODEOWNERS:1]: missed that one to delete ...

File: packages/matter-node.js/src/codec/BitmapSchema.ts (hunk 3 above)
Comment (Apollon77) [reply packages/matter-node.js/src/codec/BitmapSchema.ts:1]: We decided to do "file merging" in a separate MR after the initial move. I will work on this once this one is in

File: tsconfig.json (hunk 4 above)
Comment (Apollon77) [reply tsconfig.json:19]: updated to how it was mened

File: packages/matter-node.js/CONTRIBUTING.md (hunk 1 above)
Comment (mfucci) [reply packages/matter-node.js/CONTRIBUTING.md:1]: OK

File: packages/matter-node.js/src/codec/BitmapSchema.ts (hunk 3 above)
Comment (mfucci) [reply packages/matter-node.js/src/codec/BitmapSchema.ts:1]: OK

**Review by mfucci — APPROVED** [review]

**Comment by Apollon77** [comment]
As agreed with @turon I will merge this one tomorrow morning, giving him some time too to look at what he now got :-)

File: matter-test-apk/app-debug.zip
Comment (turon) [inline matter-test-apk/app-debug.zip:1]: Why do we need such a binary?

File: packages/matter-node.js/CONTRIBUTING.md (hunk 1 above)
Comment (turon) [reply packages/matter-node.js/CONTRIBUTING.md:1]: Really we should have the packages mirror the following top-level files:
```
CONTRIBUTING.md
CODE_OF_CONDUCT.md
LICENSE
```

Any unique info in this file should be moved to the README, moved to some newly named file (`BUILDING.md`?), or copied to the end of the CSA `CONTRIBUTING.md` boiler plate.

File: packages/matter-node.js/src/codec/Base38.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
+/**
```
Comment (turon) [inline packages/matter-node.js/src/codec/Base38.ts:1]: These codecs should move to matter.js.

File: packages/matter-node.js/src/crypto/Crypto.ts [hunk 6]
```diff
@@ -0,0 +1,195 @@
+/**
```
Comment (turon) [inline packages/matter-node.js/src/crypto/Crypto.ts:1]: Crypto libs should move to matter.js

File: packages/matter.js/src/cluster/GeneralCommissioningCluster.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 /**
  * @license
- * Copyright 2022 Project CHIP Authors
+ * Copyright 2022-2023 Project CHIP Authors
```
Comment (turon) [inline packages/matter.js/src/cluster/GeneralCommissioningCluster.ts:3]: These one line copyright updates aren't strictly necessary, but they don't hurt either.  Is the linter mandating this, or is it just precision?

**Review by turon — APPROVED** [review]
:+1: LGTM

File: .eslintrc.json [hunk 8]
```diff
@@ -38,10 +38,12 @@
             {
                 "ignoreRestSiblings": true,
                 "argsIgnorePattern": "^_",
-                "varsIgnorePattern": "^Matter.*SpecificationV"
+                "varsIgnorePattern": "^Matter.*SpecificationV|^_"
```
[the remark this answers is not in the document]
Comment (Apollon77) [reply .eslintrc.json:41]: it is the "standard convention" if you allow "unused variables to be specified" thet thats ok when they have an _ at the beginning. I would remove that once I cleared up some places and then only allow it for "unused parameters".

File: matter-test-apk/app-debug.zip
Comment (Apollon77) [reply matter-test-apk/app-debug.zip:1]: @mfucci added had it in node-matter ... we can also remove later. I do not use it

File: packages/matter-node.js/src/codec/Base38.ts (hunk 5 above)
Comment (Apollon77) [reply packages/matter-node.js/src/codec/Base38.ts:1]: Yes will be one of the next merge steps in extra PRs as planned

File: packages/matter-node.js/src/crypto/Crypto.ts (hunk 6 above)
Comment (Apollon77) [reply packages/matter-node.js/src/crypto/Crypto.ts:1]: Yes - and will also get abstracted. On the plan

File: packages/matter.js/src/cluster/GeneralCommissioningCluster.ts (hunk 7 above)
Comment (Apollon77) [reply packages/matter.js/src/cluster/GeneralCommissioningCluster.ts:3]: just precision ... Having such an update once a year is ok I think ... or we remove it, but having a mixure of different year infos might be strange

**Comment by Apollon77** [comment]
For Contibuting: Ok I now copied the files and added a todo to merge CONTRIBUTING ... will be own PR

File: packages/matter-node.js/CONTRIBUTING.md (hunk 1 above)
Comment (Apollon77) [reply packages/matter-node.js/CONTRIBUTING.md:1]: I now copied the LICENSE and CODE_OF_CONDUCT.md also in the respective packages and added a TODO for myself to merge(sync/split CONTRIBUTING.md in matter-node.js as a follow up PR

**Pull request merged**